tests/unit_tests/azure/test_cloud.py

```diff
@@ -119,3 +130,366 @@ def test_create_network_interface_with_inbound_ports(
             expected_calls
             == network_group_mock.begin_create_or_update.call_args_list
         )
+
+
+@mock.patch(
+    "pycloudlib.azure.util.get_client",
+    side_effect=[
+        resource_client_mock,
+        network_client_mock,
+        compute_client_mock,
+    ],
+)
+class TestNonComputeParamsOverrides:
+    def test_rg_params_override(self, _m_get_client):
+        type(resource_mock).name = mock.PropertyMock(
+            return_value="resource_group"
+        )
+        type(resource_client_mock).resource_groups = mock.PropertyMock(
+            return_value=resource_group_mock
+        )
+        resource_group_mock.create_or_update.return_value = resource_mock
+        resource_group_mock.get.side_effect = ResourceNotFoundError()
+
+        new_params = {"location": "new_location", "tags": {"name": "new-tag"}}
+        new_rg = AzureParams("default-pyc-rg", new_params)
+        instance = Azure(
```

Comment:
Being Azure as subclass of BaseCloud, this is typically named as cloud because it represents that concept. We also have BaseInstance, which represent instances within a particular cloud. I think it makes sense to be consistent and name this as cloud in this and the other tests.

```suggestion
        cloud = Azure(
```